packages/cli/src/build.test.ts

```diff
@@ -352,137 +412,44 @@ describe('build', () => {
 
   describe('project references', () => {
     describe('node 10', () => {
-      const log = vi.spyOn(console, 'log').mockImplementation(noOp);
-      let files: Record<string, string>;
-
-      beforeAll(() => {
-        files = compile(
-          getFixture('project-references-node-10'),
-          ['commonjs', 'module'],
-          {
-            references: true,
-            verbose: true,
-          },
-        );
-      });
-
-      it('builds all projects when using project references', () => {
-        expect(Object.keys(files)).toStrictEqual([
-          'packages/project-3/dist/foo.mjs',
-          'packages/project-3/dist/foo.d.mts.map',
-          'packages/project-3/dist/foo.d.mts',
-          'packages/project-3/dist/index.mjs',
-          'packages/project-3/dist/index.d.mts.map',
-          'packages/project-3/dist/index.d.mts',
-          'packages/project-3/tsconfig.tsbuildinfo',
-          'packages/project-3/dist/foo.cjs',
-          'packages/project-3/dist/foo.d.cts.map',
-          'packages/project-3/dist/foo.d.cts',
-          'packages/project-3/dist/index.cjs',
-          'packages/project-3/dist/index.d.cts.map',
-          'packages/project-3/dist/index.d.cts',
-          'packages/project-1/dist/index.mjs',
-          'packages/project-1/dist/index.d.mts.map',
-          'packages/project-1/dist/index.d.mts',
-          'packages/project-1/tsconfig.tsbuildinfo',
-          'packages/project-1/dist/index.cjs',
-          'packages/project-1/dist/index.d.cts.map',
-          'packages/project-1/dist/index.d.cts',
-          'packages/project-2/dist/index.mjs',
-          'packages/project-2/dist/index.d.mts.map',
-          'packages/project-2/dist/index.d.mts',
-          'packages/project-2/tsconfig.tsbuildinfo',
-          'packages/project-2/dist/index.cjs',
-          'packages/project-2/dist/index.d.cts.map',
-          'packages/project-2/dist/index.d.cts',
-          'dist/index.mjs',
-          'dist/index.d.mts.map',
-          'dist/index.d.mts',
-          'dist/index.cjs',
-          'dist/index.d.cts.map',
-          'dist/index.d.cts',
-        ]);
-      });
-
-      it('logs the project references', () => {
-        expect(log).toHaveBeenCalledWith(
-          expect.stringMatching(
-            /Building referenced project ".*packages\/project-1\/tsconfig\.json.*"\./u,
+      vi.spyOn(console, 'log').mockImplementation(noOp);
+
+      it('builds all projects and logs the project references', async () => {
```

Comment:
This is the only thing I haven't been able to resolve. Previously we tested this by hooking into the `writeFile` TypeScript system function, but since this is now called inside a worker, we can't mock it.

I was considering sending messages from the worker to the main thread with a list of emitted files, but maybe you have better ideas?

Reply:
The other option would be to write an integration / e2e test and run the actual CLI on an actual project in the actual file system. A suggestion for how to configure Vitest to do this lives [here](https://github.com/vitest-dev/vitest/discussions/4675#discussioncomment-7775007).

Reply:
For present purposes, I'll settle for an issue and we can do it later?

Reply:
#60